integration/test/Notifications.hs

```diff
@@ -177,9 +189,36 @@ isUserConnectionNotif = notifTypeIsEqual "user.connection"
 
 isConnectionNotif :: MakesValue a => String -> a -> App Bool
 isConnectionNotif status n =
-  (&&)
-    <$> nPayload n %. "type" `isEqual` "user.connection"
-    <*> nPayload n %. "connection.status" `isEqual` status
+  nPayload n %. "type" `isEqual` "user.connection"
+    &&~ nPayload n %. "connection.status" `isEqual` status
+
+-- NB:
+-- (&&) <$> (print "hello" *> pure False) <*> fail "bla" === _|_
+-- runMaybeT $  (lift (print "hello") *> MaybeT (pure Nothing)) *> lift (fail "bla") === pure Nothing
+
+-- | make Bool lazy
+liftBool :: Functor f => f Bool -> BoolT f
+liftBool = MaybeT . fmap (bool Nothing (Just ()))
+
+-- | make Bool strict
+unliftBool :: Functor f => BoolT f -> f Bool
+unliftBool = fmap isJust . runMaybeT
+
+-- | lazy (&&)
+(&&~) :: App Bool -> App Bool -> App Bool
```

Comment:
it's fine to introduce this locally, but you probably shouldn't export it, then?

Reply:
I think it makes sense to export it, it may be needed elsewhere? I would also encourage users to use this instead of `liftA2 (&&)`

Reply:
fine, but then please don't export it from `/integration/test/Notifications.hs`.

Reply:
Alright!